docs/developer/architecture/user_customization_and_persistence.rst

```diff
@@ -0,0 +1,214 @@
+User Customization & Persistence
+================================
+
+Context
+-------
+
+.. todo::
+
+   This section will be handled by the requirements PR.
+
+
+Design Choices
+--------------
+
+:term:`ConfigMap` KV(Key-Value) store is chosen as a unified data access and
+storage media for user editable configurations in a MetalK8s cluster based on
+the above requirements for the following reasons:
+
+* The limited number of read/write operations even for multi-node setups means
+  that concurrency issues on CRUD operations will not lag even if we do not
+  implement cross-app locking.
+* Ability to support CRUD operations on ConfigMap's with CLI and UI easily
+  using our already existing python kubernetes module.
+* Guarantee of adaptability and ease of changing the design and implementation
+  in cases were customer needs evolve rapidly(modular design nature)
+* Ease of operation and a guarantee of no downtime in a MetalK8s cluster
+  especially when configuration changes are made on the fly.
```

Comment:
How does using ConfigMaps handle "no-downtime" guarantees?

Reply:
> changes are made on the fly.

WDYM? Do you expect a change to impact the target service right away? Or not?
"made on the fly" would mean something like `kubectl edit cm ...` or something else?

Reply:
Well, this was supposed to mean; the fact that we are making use of ConfigMaps, I think once a user updates a given configuration, we do not require Pod restarts for the specific services to pick up the changes. no?